src/org/labkey/targetedms/parser/PrecursorChromInfo.java

```diff
@@ -445,71 +445,86 @@ public Chromatogram createChromatogram(TargetedMSRun run, boolean loadFromSkyd)
 
             ChromatogramBinaryFormat binaryFormat = ChromatogramBinaryFormat.values()[getChromatogramFormat()];
 
-            byte[] databaseBytes = getChromatogram();
-            byte[] compressedBytes = databaseBytes;
+            Object[] compressedBytesAndStatus = getCompressedBytesAndStatus(run, loadFromSkyd);
+            byte[] compressedBytes = (byte[]) compressedBytesAndStatus[0];
+            Chromatogram.SourceStatus status = (Chromatogram.SourceStatus) compressedBytesAndStatus[1];
 
-            Chromatogram.SourceStatus status;
+            if (compressedBytes == null)
+            {
+                return null;
+            }
 
-            if (loadFromSkyd || databaseBytes == null)
+            byte[] uncompressedBytes = SkylineBinaryParser.uncompressStoredBytes(compressedBytes, getUncompressedSize(), _numPoints, _numTransitions);
+            return binaryFormat.readChromatogram(uncompressedBytes, _numPoints, _numTransitions, status);
+        }
+        catch (IOException | DataFormatException exception)
+        {
+            throw new UnexpectedException(exception);
+        }
+    }
+
+    private Object[] getCompressedBytesAndStatus(TargetedMSRun run, boolean loadFromSkyd)
```

Comment:
Let's make a class to hold the return value here, or at least use a Pair<byte[], Chromatogram.SourceStatus> instead of an untyped Object array.

Reply:
I added a class to hold the return values.